earth_enterprise/src/fusion/autoingest/AssetSerializer.h

```diff
@@ -14,13 +18,81 @@ class AssetSerializerInterface {
     virtual ~AssetSerializerInterface() = default;
 };
 
+/*
+* GetFileInfo and ReadXMLDocument were added as function templates to allow
+* them to be overridden in unit tests. The versions here should be used for
+* all asset types during normal operation.
+*/
+template<class AssetType>
```

Comment:
It looks like these don't need to be templates. If you remove the template header and mark them `inline` I think they will work fine.

Reply:
Update: maybe they need to be templates so you can override them in the unit tests. I guess I missed that in the comment.

Reply:
Yeah, if not for the unit tests, those wouldn't need to be separate functions at all, template or otherwise. I pulled them out like that so that I wouldn't need to create temporary files for `GetFileInfo` and `ReadDocument` to act on.
It does add some otherwise unnecessary complexity, so if there's a better way to go about it, I'm certainly open to it.